util/src/main/java/tech/pegasys/pantheon/util/enode/EnodeURL.java

```diff
@@ -25,19 +25,14 @@
 
 import com.google.common.base.Objects;
 import com.google.common.net.InetAddresses;
+import com.google.common.primitives.Ints;
 
 public class EnodeURL {
 
-  private static final String HEX_STRING_PATTERN = "[0-9a-fA-F]+";
-
-  private static final String ENODE_URL_PATTERN_NEW =
-      "^enode://"
-          + "(?<nodeId>\\w+)"
-          + "@"
-          + "(?<ip>.*)"
-          + ":"
-          + "(?<listening>\\d+)"
-          + "(\\?discport=(?<discovery>\\d+))?$";
+  private static final int NODE_ID_SIZE = 64;
+  private static final Pattern DISCPORT_QUERY_STRING_REGEX =
+      Pattern.compile("discport=([0-9]{1,5})");
```

Comment:
If we're being really strict this should probably be `"(^|&)discport=([0-9]{1,4})"` I think. So discport should either be the first arg or after a & to prevent matching `enode://...@...:30303?notthediscport=1234`

Reply:
Made this check stricter and added more tests.  For now, I have the parsing fail if there are any additional params outside of a singular discport param.  It's simpler to implement this way and if we need additional params for some reason we can add that later.